tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -218,5 +228,39 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            var logDaysLimit = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.MaxLogFilesDays);
+            if (int.TryParse(logDaysLimit, out var days) && (days > 0 && days < int.MaxValue))
+            {
+                var date = DateTime.Now.AddDays(-1 * days);
+                var logFormats = new[]
+                  {
+                    "dotnet-tracer-managed-*-.log",
+                    "dotnet-tracer-native-*-.log",
+                    "dotnet-native-loader-*.log",
+                    "DD-DotNet-Profiler-Native-*.log"
+                  };
+
+                foreach (var logFormat in logFormats)
+                {
+                    foreach (var logFile in Directory.EnumerateFiles(logsDirectory, logFormat))
+                    {
+                        if (File.GetLastWriteTime(logFile) < date)
+                        {
```

Comment:
Probably worth putting the `if()` inside the try-catch too, as that's an IO operation (e.g. if the user deletes the file manually while this is running we would abort.

Although, maybe it's a _good_ idea to abort early. What are the chances that there's only 1 file that causes errors? It's more likely we can't access _any_ files, and we don't want to be generating 30 exceptions in that case I guess? 🤔 So maybe it's a good idea to abandon on the first exception? What do you think?

Reply:
Yeah I think that makes sense, the other thing I think of is what if for some reason a single file is open for example and that one yields the exception but the others are good to access? Should we take this possibility into account or is safer to abandon earlier and if anything next time the user deploys the function will run again?